### PR TITLE
update docs for config file

### LINKED ISSUE #116: Document config file
See title (and original thread)

_Originally posted by @ryanmrichard in https://github.com/CMakePP/CMinx/pull/104#discussion_r963739248_

### PR DESCRIPTION
**Pull request by ryanmrichard**
**Is this pull request associated with an issue(s)?**
Fixes #116.

**Description**
This PR adds a new documentation page describing the configuration file syntax, how to write
a configuration file, and how to tell CMinx to use your configuration file. I also updated
the page showing the usage information since it was out of date.

**TODOs**
None. This is r2g.

### CODE REVIEW

File: docs/source/config.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+``~/.config/cminx/``). See
+`here <https://confuse.readthedocs.io/en/latest/usage.html#search-paths>`__ for
+more details regarding how Confuse locates configuration files.
+
+For organizational and namespacing purposes we have partitioned the
+options recognized by CMinx into four categories:
+
+1. ``input``.   Affects which files and file parts CMinx will parse.
+2. ``output``.  Affects how CMinx outputs results.
+3. ``logging``. Options pertaining to CMinx's log messages.
```
Comment (AutonomicPerfectionist) [inline docs/source/config.rst:45]: Should probably mention that this takes any options recognized by Python's logging configuration system, probably add a link to [Logging config Dict Schema](https://docs.python.org/3/library/logging.config.html#logging-config-dictschema) and explain that the Yaml section is considered analogous to a dictionary

**Review by AutonomicPerfectionist — CHANGES_REQUESTED** [review]
@ryanmrichard for the most part this looks good but I noticed there wasn't much said about the logging section. Since that section can hold any valid Python Logging module configuration I figured we should probably say something about ti

File: docs/source/config.rst (hunk 1 above)
Comment (ryanmrichard) [reply docs/source/config.rst:45]: Would you be able to write it? I'm not actually familiar with logging module configurations.

Comment (AutonomicPerfectionist) [inline docs/source/config.rst:45]: > `3. ``logging``. Options pertaining to CMinx's log messages.`

Suggested replacement: 
```rst
3. ``logging``. Options pertaining to CMinx's log messages. See :ref:`logging_options` for more details.

```

File: docs/source/config.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+directly to the ``main`` function like:
+
+.. code-block:: Python
+
+   import cminx
+   cminx.main(['-s', 'config.yaml', <other options>])
+
+(in both code examples ``<other options>`` should be replaced with any
+additional command line options the user wants to pass to CMinx).
+
```
Comment (AutonomicPerfectionist) [inline docs/source/config.rst:85]: Suggestions: add the following before `.. _default_config:`

```rst

.. _logging_options:

*******************
Logging Options
*******************
In CMinx's configuration files, there is a `logging` section. This section is special in that
it forwards all options inside of it to the Python `logging.config` module. This means
that users can configure the Python logging system, and thus CMinx's logging settings,
directly from a standard config file.

See `here <https://docs.python.org/3/library/logging.config.html#logging-config-dictschema>`__
for all valid logging configuration options. The YAML examples present within will work
as long as they are within the `logging` section of the config file.
See :ref:`default_config` for an example of a basic logging configuration.


```

**Review by AutonomicPerfectionist — CHANGES_REQUESTED** [review]
@ryanmrichard I don't have direct write access so I just put my suggestions in review comments

File: docs/source/config.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+.. _logging_options:
+
+*******************
+Logging Options
+*******************
+In CMinx's configuration files, there is a `logging` section. This section is
+special in that it forwards all options inside of it to the Python
+`logging.config` module. This means that users can configure the Python logging
+ system, and thus CMinx's logging settings, directly from a standard config
+ file.
```
Comment (AutonomicPerfectionist) [inline docs/source/config.rst:97]: There appears to be extra indentations on these lines that cause the generated HTML to look a bit funny

File: docs/source/config.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+See
+`here <https://docs.python.org/3/library/logging.config.html#logging-config-dictschema>`__
+for all valid logging configuration options. The YAML examples present within
+will work as long as they are within the `logging` section of the config file.
+See :ref:`default_config` for an example of a basic logging configuration.
```
Comment (AutonomicPerfectionist) [inline docs/source/config.rst:103]: I forgot to add the `:code:` role to the interpreted text, I would add that in front of the `logging` and `logging.config` mentions to make them appear better in the generated HTML

**Review by AutonomicPerfectionist — CHANGES_REQUESTED** [review]
@ryanmrichard got just two more changes to be made, mainly with the stuff I just suggested. There's a funky indentation error and then I forgot to add the `:code:` role for some inline interpreted text, Should be good after those are fixed though

File: docs/source/config.rst (hunk 4 above)
Comment (ryanmrichard) [reply docs/source/config.rst]: These should be fixed. I also updated the CMakePP organization's team permissions so in the future you should have write access.

**Review by AutonomicPerfectionist — APPROVED** [review]
@ryanmrichard looks good now

**Comment by ryanmrichard** [comment]
🚀 [[bumpr]](https://github.com/haya14busa/action-bumpr) [Bumped!](https://github.com/CMakePP/CMinx/actions/runs/3023104309) **New version**:[v1.0.11](https://github.com/CMakePP/CMinx/releases/tag/v1.0.11) **Changes**:[v1.0.10...v1.0.11](https://github.com/CMakePP/CMinx/compare/v1.0.10...v1.0.11)

**Pull request merged**